src/models/Customer.java

```diff
@@ -24,7 +28,29 @@ public String getFullName() {
 	public String getPhoneNum() {
 		return phoneNum;
 	}
-	
-	
 
+	public void setEmailAddress(String emailAddress) throws InvalidInputException {
```

Comment:
Please change the Constructor to use setter!
like
```
Customer () {
this.name = name -> setName(name)
```

Reply:
And I believe you should handle the throws in inside or outside of Constructor.
Wrap codes try / catch, and make it safer!